src/route-recognizer.js

```diff
@@ -334,9 +346,12 @@ function findHandler(state, path, queryParams) {
   return result;
 }
 
-function addSegment(currentState, segment) {
+function addSegment(currentState, segment, caseSensitive) {
   let state = currentState;
   segment.eachChar(ch => {
+    if (!caseSensitive && segment instanceof StaticSegment) {
+      ch.validChars = ch.validChars.toUpperCase() + ch.validChars.toLowerCase();
+    }
```

Comment:
It'd be cleaner for the `StaticSegment` ctor to receive the `caseSensitive` value and adjust its own `validChars`.